homeassistant/components/growatt_server/sensor.py

```diff
@@ -105,6 +91,11 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     add_entities(entities, True)
 
 
+async def async_setup_entry(hass, config_entry, add_entities):
+    """Set up growatt server from Config Flow."""
+    hass.async_add_executor_job(setup_platform, hass, config_entry.data, add_entities)
```

Comment:
We should send an empty dict as discovery_info to `setup_platform` and add a guard clause there that checks if discovery_info is None and return if so.

This will avoid setting up the platform from a platform config section.